src/reanimated2/shareables.ts

```diff
@@ -241,14 +241,28 @@ Offending code was: \`${getWorkletCode(value)}\``);
         shareableMappingCache.set(value, inaccessibleObject);
         return inaccessibleObject;
       }
-      if (__DEV__) {
-        // we freeze objects that are transformed to shareable. This should help
-        // detect issues when someone modifies data after it's been converted to
-        // shareable. Meaning that they may be doing a faulty assumption in their
+      if (__DEV__ && !isHostObject(value) && isPlainJSObject(value)) {
+        // We freeze objects that are transformed to Shareable with a meaningful warning.
+        // This should help detect issues when someone modifies data after it's been converted to
+        // Shareable. Meaning that they may be doing a faulty assumption in their
         // code expecting that the updates are going to automatically populate to
         // the object sent to the UI thread. If the user really wants some objects
-        // to be mutable they should use shared values instead.
-        Object.freeze(value);
+        // to be mutable they should use Shared Values instead.
+        Object.entries(value).forEach(([key, element]) => {
+          Object.defineProperty(value, key, {
+            get() {
+              return element;
+            },
+            set() {
+              console.warn(
+                `[Reanimated] Tried to modify key \`${key}\` of an object which has been converted to a Shareable. This will not affect worklets which have already received the object. If you want this change to be reflected in the worklets, consider using a Shared Value instead. See 
```

Comment:
Shareable is an implementation details, let's not leak internals into error messages.
```suggestion
                `[Reanimated] Tried to modify key \`${key}\` of an object which has been already passed to a worklet. The new value will not be propagated to worklets which have already received the object which may lead to inconsistencies. If you want this change to be reflected in the worklets, consider using a shared value instead. See 
```
Also, let's just provide a link to troubleshooting instead of shooting the full solution here.